src/routes.js

```diff
@@ -637,26 +641,27 @@ const routes = [
           fullWidth: true
         }
       }),
-      asyncRoute({
-        path: 'analytics',
-        component: 'AnalyticsList',
-        loading: 'tab',
-        meta: {
-          validateData: {
-            currentUser: () => currentUser.can([
-              'config.read',
-              'config.set',
-              'analytics.read'
-            ])
-          },
-          title: () => [
-            i18n.t('systemHome.tab.analytics'),
-            i18n.t('systemHome.title')
-          ],
-          fullWidth: true
-        },
-        beforeEnter: () => (config.showsAnalytics ? true : '/')
-      })
+      ...conditionalRoutes(config.showsAnalytics, [
```

Comment:
I think what's on `master` works OK for /system/analytics, but I wanted to take a different approach for /reset-password and /account/claim. I left some notes in the commit message for 7335412ca6fbf419e413ac6e7ad0a127b4c51ee2:

> Making this change because different disabled routes were behaving
slightly differently depending on their meta fields. That's because
global beforeEach guards are called before a beforeEnter guard in the
route config. For example, navigating to /reset-password sent a request
to restore the session, while navigating to /account/claim did not. Now
disabled routes are removed entirely, so all resolve to the NotFound
route.

After introducing `conditionalRoutes()` for /reset-password and /account/claim, I also used `conditionalRoutes()` for /system/analytics so that the same approach is used for all three routes.

Reply:
I like this new approach, looks cleaner